xbmc/utils/SeekHandler.cpp

```diff
@@ -182,6 +182,33 @@ void CSeekHandler::Seek(bool forward, float amount, float duration /* = 0 */, bo
   m_timer.StartZero();
 }
 
+void CSeekHandler::SeekSeconds(int seconds)
+{
+  if (seconds == 0)
+    return;
+
+  m_requireSeek = true;
+  g_infoManager.SetSeeking(true);
+  m_seekStep = 0;
+  m_analogSeek = false;
+  m_seekDelay = 0;
+
+  if (g_infoManager.GetTotalPlayTime())
+    m_percentPlayTime = (float)g_infoManager.GetPlayTime() / g_infoManager.GetTotalPlayTime() * 0.1f;
+  else
+    m_percentPlayTime = 0.0f;
+
+  float percentPerSecond = 0.0f;
+  if (g_infoManager.GetTotalPlayTime())
+    percentPerSecond = 100.0f / (float)g_infoManager.GetTotalPlayTime();
+
+  m_percent = m_percentPlayTime + percentPerSecond * seconds;
+  g_infoManager.SetSeekStepSize(seconds);
+
+  if (m_percent > 100.0f) m_percent = 100.0f;
```

Comment:
please don't place multiple statements in a single line. this style is really ugly if you want to set a breakpoint